### PR TITLE
Sona :Fix sonar bugs and code smells in iotdb-client 

### PR DESCRIPTION
**Pull request by 2b3c511**
This pr aims to fix sonar bugs and code smells in iotdb-client, including the following packages:
cli:
iotdb-client/cli/src/main/java/org/apache/iotdb/cli/AbstractCli.java
iotdb-client/cli/src/main/java/org/apache/iotdb/cli/IoTDBSyntaxHighlighter.java
iotdb-client/cli/src/main/java/org/apache/iotdb/cli/utils/IoTPrinter.java
iotdb-client/cli/src/main/java/org/apache/iotdb/cli/utils/JlineUtils.java
iotdb-client/cli/src/main/java/org/apache/iotdb/tool/AbstractCsvTool.java
iotdb-client/cli/src/main/java/org/apache/iotdb/tool/AbstractTsFileTool.java
iotdb-client/cli/src/main/java/org/apache/iotdb/tool/ExportCsv.java
iotdb-client/cli/src/main/java/org/apache/iotdb/tool/ExportTsFile.java
iotdb-client/cli/src/main/java/org/apache/iotdb/tool/ImportCsv.java
iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTests.java
isession:
iotdb-client/isession/src/main/java/org/apache/iotdb/isession/ISession.java
iotdb-client/isession/src/main/java/org/apache/iotdb/isession/SessionConfig.java
iotdb-client/isession/src/main/java/org/apache/iotdb/isession/SessionDataSet.java
iotdb-client/isession/src/main/java/org/apache/iotdb/isession/pool/ISessionPool.java
iotdb-client/isession/src/main/java/org/apache/iotdb/isession/template/Template.java
iotdb-client/isession/src/main/java/org/apache/iotdb/isession/template/TemplateNode.java
jdbc:
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/AbstractIoTDBJDBCResultSet.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/Activator.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/Config.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/GroupedLSBWatermarkEncoder.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBConnection.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBDataSource.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBDatabaseMetadata.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBDriver.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBJDBCResultSet.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBNonAlignJDBCResultSet.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBPreparedStatement.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBResultMetadata.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBStatement.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBTracingInfo.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/StringUtils.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/Utils.java
iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/WatermarkEncoder.java
iotdb-client/jdbc/src/test/java/org/apache/iotdb/jdbc/IoTDBPreparedStatementTest.java
iotdb-client/jdbc/src/test/java/org/apache/iotdb/jdbc/UtilsTest.java
service-rpc:
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/AutoResizingBuffer.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/AutoScalingBufferReadTransport.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/AutoScalingBufferWriteTransport.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/BatchExecutionException.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/ConfigurableTByteBuffer.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/IoTDBJDBCDataSet.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/IoTDBRpcDataSet.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/RpcTransportFactory.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/StatementExecutionException.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/TConfigurationConst.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/TNonblockingSocketWrapper.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/TSStatusCode.java
iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/TSocketWrapper.java
session:
iotdb-client/session/src/main/java/org/apache/iotdb/session/InsertConsumer.java
iotdb-client/session/src/main/java/org/apache/iotdb/session/Session.java
iotdb-client/session/src/main/java/org/apache/iotdb/session/SessionConnection.java
iotdb-client/session/src/main/java/org/apache/iotdb/session/template/InternalNode.java
iotdb-client/session/src/main/java/org/apache/iotdb/session/util/SessionUtils.java
iotdb-client/session/src/main/java/org/apache/iotdb/session/util/ThreadUtils.java
iotdb-client/session/src/test/java/org/apache/iotdb/session/SessionCacheLeaderUTTest.java

### CODE REVIEW

File: iotdb-client/cli/src/main/java/org/apache/iotdb/cli/Cli.java [hunk 1]
```diff
@@ -173,22 +176,9 @@ private static void receiveCommands(LineReader reader) throws TException {
...
-          try {
-            reader.readLine("Press CTRL+C again to exit, or press ENTER to continue", '\0');
-          } catch (UserInterruptException | EndOfFileException e2) {
-            System.exit(CODE_OK);
-          }
-        } catch (EndOfFileException e) {
-          // Exit on EOF (usually by pressing CTRL+D).
-          System.exit(CODE_OK);
+        boolean readLine = readerReadLine(reader, connection);
+        if (Boolean.TRUE.equals(readLine)) {
```
Comment (Beyyes) [inline iotdb-client/cli/src/main/java/org/apache/iotdb/cli/Cli.java:180]: why use `Boolean.TRUE.equals`

File: iotdb-client/cli/src/main/java/org/apache/iotdb/cli/utils/JlineUtils.java [hunk 2]
```diff
@@ -83,10 +85,6 @@ public static LineReader getLineReader(String username, String host, String port
             + username.hashCode();
     builder.variable(LineReader.HISTORY_FILE, new File(historyFilePath));
 
-    // TODO: since the lexer doesn't produce tokens for quotation marks, disable the highlighter to
```
Comment (Beyyes) [inline iotdb-client/cli/src/main/java/org/apache/iotdb/cli/utils/JlineUtils.java:86]: leave `TODO`

File: iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+public class WriteCsvFileTest {
+  @Test
+  public void writeCsvFileTest() {
+    List<String> headerNames =
+        new ArrayList<>(Arrays.asList("Time", "column1", "column2", "column3"));
```
Comment (Beyyes) [inline iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTest.java:36]:
```suggestion
        Arrays.asList("Time", "column1", "column2", "column3");
```

File: iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    List<Object> row1 = new ArrayList<>(Arrays.asList(1, null, "hello,world", true));
```
Comment (Beyyes) [inline iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTest.java:38]:
```suggestion
    List<Object> row1 = Arrays.asList(1, null, "hello,world", true);
```

File: iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    List<Object> row2 = new ArrayList<>(Arrays.asList(2, "", "hello,world", false));
```
Comment (Beyyes) [inline iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTest.java:39]: Arrays.asList is enough?

File: iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/AbstractIoTDBJDBCResultSet.java [hunk 6]
```diff
@@ -452,7 +455,7 @@ public ResultSetMetaData getMetaData() {
         nonAlign = true;
       }
     } catch (SQLException throwables) {
-      throwables.printStackTrace();
+      LOGGER.error(String.format("get meta data error:%s", throwables.getMessage()));
```
Comment (Beyyes) [inline iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/AbstractIoTDBJDBCResultSet.java:458]:
```suggestion
      LOGGER.error("get meta data error:{}", throwables.getMessage());
```

File: iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBDataSource.java [hunk 7]
```diff
@@ -87,7 +91,7 @@ public Connection getConnection() throws SQLException {
     try {
       return new IoTDBConnection(url, properties);
     } catch (TTransportException e) {
-      e.printStackTrace();
+      logger.error(String.format("get connection error : %s", e.getMessage()));
```
Comment (Beyyes) [inline iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBDataSource.java:94]:
```suggestion
      logger.error("get connection error : {}", e.getMessage());
```

File: iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBDataSource.java [hunk 8]
```diff
@@ -100,7 +104,7 @@ public Connection getConnection(String username, String password) {
       newProp.setProperty(PWD_STR, password);
       return new IoTDBConnection(url, newProp);
     } catch (Exception e) {
-      e.printStackTrace();
+      logger.error(String.format("get connection error : %s", e.getMessage()));
```
Comment (Beyyes) [inline iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBDataSource.java:107]:
```suggestion
      logger.error("get connection error : {}", e.getMessage());
```

File: iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/StringUtils.java [hunk 9]
```diff
@@ -47,8 +52,9 @@ public class StringUtils {
       byteToChars[j] = allBytesString.charAt(j);
     }
     try {
-      toPlainStringMethod = BigDecimal.class.getMethod("toPlainString", new Class[0]);
+      toPlainStringMethod = BigDecimal.class.getMethod("toPlainString");
     } catch (NoSuchMethodException nsme) {
+      LOGGER.info(String.format("to plain String method Error: %s", nsme));
```
Comment (Beyyes) [inline iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/StringUtils.java:57]:
```suggestion
      LOGGER.info(String.format("To plain String method Error: %s", nsme));
```

File: iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/TConfigurationConst.java [hunk 10]
```diff
@@ -23,9 +23,11 @@
 
 public class TConfigurationConst {
   // https://github.com/apache/thrift/blob/master/doc/specs/thrift-tconfiguration.md
-  public static TConfiguration defaultTConfiguration =
+  public static final TConfiguration defaultTConfiguration =
```
Comment (Beyyes) [inline iotdb-client/service-rpc/src/main/java/org/apache/iotdb/rpc/TConfigurationConst.java:26]:
```suggestion
  public static final TConfiguration DEFAULT_TCONFIGURATION =
```

File: iotdb-client/session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java [hunk 11]
```diff
@@ -496,11 +529,6 @@ private void putBack(ISession session) {
     synchronized (this) {
       // we do not need to notifyAll as any waited thread can continue to work after waked up.
       this.notify();
-      // comment the following codes as putBack is too frequently called.
```
Comment (Beyyes) [inline iotdb-client/session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java:499]: leave this TODO?

File: iotdb-client/cli/src/main/java/org/apache/iotdb/tool/ImportCsv.java [hunk 12]
```diff
@@ -780,11 +798,16 @@ private static void parseHeaders(
     String regex = "(?<=\\()\\S+(?=\\))";
     Pattern pattern = Pattern.compile(regex);
     for (String headerName : headerNames) {
-      if ("Time".equalsIgnoreCase(headerName)) {
-        timeColumn = headerName;
-        continue;
-      } else if ("Device".equalsIgnoreCase(headerName)) {
-        deviceColumn = headerName;
+
+      boolean ignoreTimeCase = "Time".equalsIgnoreCase(headerName);
```
Comment (Beyyes) [inline iotdb-client/cli/src/main/java/org/apache/iotdb/tool/ImportCsv.java:802]: why change this

File: iotdb-client/cli/src/main/java/org/apache/iotdb/cli/Cli.java (hunk 1 above)
Comment (2b3c511) [reply iotdb-client/cli/src/main/java/org/apache/iotdb/cli/Cli.java]: done

File: iotdb-client/cli/src/main/java/org/apache/iotdb/cli/utils/JlineUtils.java (hunk 2 above)
Comment (2b3c511) [reply iotdb-client/cli/src/main/java/org/apache/iotdb/cli/utils/JlineUtils.java]: done

File: iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTest.java (hunk 3 above)
Comment (2b3c511) [reply iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTest.java]: done

File: iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTest.java (hunk 5 above)
Comment (2b3c511) [reply iotdb-client/cli/src/test/java/org/apache/iotdb/tool/unit/WriteCsvFileTest.java]: done

File: iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/StringUtils.java (hunk 9 above)
Comment (2b3c511) [reply iotdb-client/jdbc/src/main/java/org/apache/iotdb/jdbc/StringUtils.java]: done

File: iotdb-client/session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java (hunk 11 above)
Comment (2b3c511) [reply iotdb-client/session/src/main/java/org/apache/iotdb/session/pool/SessionPool.java]: done

File: iotdb-client/cli/src/main/java/org/apache/iotdb/tool/ImportCsv.java (hunk 12 above)
Comment (2b3c511) [reply iotdb-client/cli/src/main/java/org/apache/iotdb/tool/ImportCsv.java]: done

**Review by Beyyes — APPROVED** [review]

**Pull request merged**